Great work @sjvrijn! Thanks. I have left a few inline comments for us to discuss. I have tested it on my machine and one test has failed... 
`FAILED tests/test_project.py::test_coverage_api_docs - AssertionError: assert '| my_python_package           | 100.00%  | 0            |' in ['Undocumente...
=1 failed, 13 passed, 2 warnings in 63.52s (0:01:03) 
`

--- a/copier.yml
+++ b/copier.yml
@@ -62,7 +62,7 @@ copyright_holder:
   default: Netherlands eScience Center
 code_of_conduct_email:
   type: str
-  default: {{ email }}
+  default: "{{ email }}"
 
 _subdirectory: template
 

--- a/copier.yml
+++ b/copier.yml
@@ -1,13 +1,5 @@
 # Essential questions
 
-directory_name:
-  type: str
-  default: my-python-project
-  help: Enter the name of the directory where the project will be created.
-  validator: >-
-        {% if not (directory_name | regex_search('^[a-z0-9\-]+$')) %}
-        directory_name must be lowercase, and can only contain letters, digits, and hyphens.
-        {% endif %}
 package_name:
   type: str
   default: my_python_package

--- a/setup.cfg
+++ b/setup.cfg
@@ -33,7 +33,7 @@ include_package_data = True
 python_requires = >=3.8
 packages =
 install_requires =
-    cookiecutter==1.7.2
+    copier==9.2.0
 
 [options.data_files]
 # This section requires setuptools>=40.6.0
@@ -45,7 +45,8 @@ install_requires =
 dev =
     coverage [toml]
     pytest
-    pytest-cookies
+    pytest-copie
+    pyprojroot
 
 
 [tool:pytest]

--- a/tests/test_project.py
+++ b/tests/test_project.py
@@ -42,15 +42,16 @@ def project_env_bin_dir(tmp_path_factory):
 
 
 @pytest.fixture(scope='session')
-def baked_with_development_dependencies(cookies_session, project_env_bin_dir):
-    result = cookies_session.bake()
-    assert result.exit_code == 0
+def baked_with_development_dependencies(tmp_path_factory, project_env_bin_dir):
+    project = run_copy(src_path=str(here()), dst_path=str(tmp_path_factory.mktemp('projects')), defaults=True)
+    project_dir = project.dst_path
+
     bin_dir = project_env_bin_dir
-    latest_pip_output = run([f'{bin_dir}python', '-m', 'pip', 'install', '--upgrade', 'pip', 'setuptools'], result.project_path)
+    latest_pip_output = run([f'{bin_dir}python', '-m', 'pip', 'install', '--upgrade', 'pip', 'setuptools'], project_dir)
     assert latest_pip_output.returncode == 0
-    pip_output = run([f'{bin_dir}python', '-m', 'pip', 'install', '--editable', '.[dev]'], result.project_path)
+    pip_output = run([f'{bin_dir}python', '-m', 'pip', 'install', '--editable', '.[dev]'], project_dir)
     assert pip_output.returncode == 0
-    return result.project_path
+    return project_dir
 
 
 def test_pytest(baked_with_development_dependencies, project_env_bin_dir):
@@ -168,7 +169,7 @@ def test_ruff_check(baked_with_development_dependencies, project_env_bin_dir):
     project_dir = baked_with_development_dependencies
     bin_dir = project_env_bin_dir
 
-    result = run([f'{bin_dir}ruff', '.'], project_dir)
+    result = run([f'{bin_dir}ruff', 'check', '.'], project_dir)
     assert result.returncode == 0
     assert '' in result.stdout
 

--- a/tests/test_project.py
+++ b/tests/test_project.py
@@ -4,18 +4,19 @@ import sys
 from sys import platform
 from typing import Sequence
 
+from pyprojroot.here import here
+from copier import run_copy
 import pytest
 
 IS_WINDOWS = platform.startswith('win')
 
 
-def test_project_folder(cookies):
-    project = cookies.bake()
+def test_project_folder(copie):
+    project = copie.copy()
 
     assert project.exit_code == 0
     assert project.exception is None
-    assert project.project_path.name == 'my-python-project'
-    assert project.project_path.is_dir()
+    assert project.project_dir.is_dir()
 
 
 def run(args: Sequence[str], dirpath: os.PathLike) -> subprocess.CompletedProcess:

